### ISSUE TITLE
multi label text classification

### ISSUE THREAD

**Issue by Jeiyoon** [SKTBrain/KoBERT#33]
안녕하세요

좋은 자료 공유해주셔서 정말 감사합니다

궁금한게 있는데 감성분석 클래스 레이블이 여러개인 데이터를 넣었을때 학습이 잘 안되는 경향을 보이는데

코드에서 레이블 갯수 외에 또 바꿔야하는 부분이 있는지 여쭤볼 수 있을까요?

정말 감사합니다!

**Comment by Jeiyoon** [comment]
현재 바꾼부분은 크게 없고 튜닝할 데이터를 추가하였고 num_classes 를 2개에서 7개로 바꿨습니다

**Comment by Jeiyoon** [comment]
https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bert.py

원래 버트에서는 BertForMultipleChoice(BertPreTrainedModel)에서 가능한 것같은데 KoBERT에서는 어떤걸 변경해야되는지 여쭤보고 싶습니다

정말 감사합니다

**Comment by Jeiyoon** [comment]
학습 진행과정은 아래와 같습니다


using cached model
using cached model
using cached model
  0%|                                                  | 0/1473 [00:00<?, ?it/s]/pytorch/torch/csrc/utils/python_arg_parser.cpp:756: UserWarning: This overload of add_ is deprecated:
	add_(Number alpha, Tensor other)
Consider using one of the following signatures instead:
	add_(Tensor other, *, Number alpha)
epoch 1 batch id 1 loss 2.023021936416626 train acc 0.140625
 14%|█████▍                                  | 200/1473 [01:00<06:14,  3.40it/s]epoch 1 batch id 201 loss 0.9267023801803589 train acc 0.5850435323383084
 27%|██████████▊                             | 400/1473 [02:00<05:16,  3.39it/s]epoch 1 batch id 401 loss 0.6681684255599976 train acc 0.6870713840399002
 41%|████████████████▎                       | 600/1473 [03:00<04:19,  3.36it/s]epoch 1 batch id 601 loss 0.31647002696990967 train acc 0.7360648918469218
 54%|█████████████████████▋                  | 800/1473 [04:00<03:21,  3.35it/s]epoch 1 batch id 801 loss 0.8212265968322754 train acc 0.7666588639200999
 68%|██████████████████████████▍            | 1000/1473 [04:59<02:20,  3.37it/s]epoch 1 batch id 1001 loss 0.00604131817817688 train acc 0.7742726023976024
 81%|███████████████████████████████▊       | 1200/1473 [05:59<01:21,  3.37it/s]epoch 1 batch id 1201 loss 0.0033375322818756104 train acc 0.787546835970025
 95%|█████████████████████████████████████  | 1400/1473 [06:59<00:22,  3.28it/s]epoch 1 batch id 1401 loss 4.904089450836182 train acc 0.7827779264810849
100%|███████████████████████████████████████| 1473/1473 [07:21<00:00,  3.34it/s]
epoch 1 train acc 0.7868614222674813
100%|███████████████████████████████████████| 1473/1473 [02:12<00:00, 11.10it/s]
epoch 1 test acc 0.0604739477257298
  0%|                                                  | 0/1473 [00:00<?, ?it/s]epoch 2 batch id 1 loss 8.59105396270752 train acc 0.125
 14%|█████▍                                  | 200/1473 [01:00<06:36,  3.21it/s]epoch 2 batch id 201 loss 0.9801695942878723 train acc 0.6480099502487562
 27%|██████████▊                             | 400/1473 [02:00<05:21,  3.33it/s]epoch 2 batch id 401 loss 0.9172695875167847 train acc 0.7184382793017456
 41%|████████████████▎                       | 600/1473 [03:00<04:18,  3.38it/s]epoch 2 batch id 601 loss 0.422923743724823 train acc 0.7479201331114809
 54%|█████████████████████▋                  | 800/1473 [04:01<03:30,  3.19it/s]epoch 2 batch id 801 loss 0.7930375933647156 train acc 0.7732911985018727
 68%|██████████████████████████▍            | 1000/1473 [05:01<02:19,  3.39it/s]epoch 2 batch id 1001 loss 0.006529033184051514 train acc 0.7746940559440559
 81%|███████████████████████████████▊       | 1200/1473 [06:00<01:20,  3.39it/s]epoch 2 batch id 1201 loss 0.005736067891120911 train acc 0.772910595337219
 95%|█████████████████████████████████████  | 1400/1473 [07:00<00:21,  3.34it/s]epoch 2 batch id 1401 loss 4.866369247436523 train acc 0.7690154354032833
100%|███████████████████████████████████████| 1473/1473 [07:22<00:00,  3.33it/s]
epoch 2 train acc 0.7737504243041412
100%|███████████████████████████████████████| 1473/1473 [02:12<00:00, 11.14it/s]
epoch 2 test acc 0.06054820095044128
  0%|                                                  | 0/1473 [00:00<?, ?it/s]epoch 3 batch id 1 loss 9.442102432250977 train acc 0.125
 14%|█████▍                                  | 200/1473 [01:00<06:14,  3.40it/s]epoch 3 batch id 201 loss 1.037582516670227 train acc 0.6412468905472637
 27%|██████████▊                             | 400/1473 [02:00<05:20,  3.35it/s]epoch 3 batch id 401 loss 0.8655645847320557 train acc 0.7150483167082294
 41%|████████████████▎                       | 600/1473 [03:00<04:16,  3.40it/s]epoch 3 batch id 601 loss 0.46280527114868164 train acc 0.7482841098169717
 54%|█████████████████████▋                  | 800/1473 [04:00<03:21,  3.34it/s]epoch 3 batch id 801 loss 0.8189800381660461 train acc 0.7733497191011236
 68%|██████████████████████████▍            | 1000/1473 [04:59<02:20,  3.37it/s]epoch 3 batch id 1001 loss 0.01821889728307724 train acc 0.7681381118881119
 81%|███████████████████████████████▊       | 1200/1473 [05:59<01:21,  3.33it/s]epoch 3 batch id 1201 loss 0.007821455597877502 train acc 0.7742246044962531
 95%|█████████████████████████████████████  | 1400/1473 [06:59<00:21,  3.33it/s]epoch 3 batch id 1401 loss 1.7888621091842651 train acc 0.7547733761598858
100%|███████████████████████████████████████| 1473/1473 [07:20<00:00,  3.34it/s]
epoch 3 train acc 0.7659220128988459
100%|███████████████████████████████████████| 1473/1473 [02:13<00:00, 11.07it/s]
epoch 3 test acc 0.0604739477257298
  0%|                                                  | 0/1473 [00:00<?, ?it/s]epoch 4 batch id 1 loss 9.867259979248047 train acc 0.125
 14%|█████▍                                  | 200/1473 [01:00<06:19,  3.36it/s]epoch 4 batch id 201 loss 0.9828888177871704 train acc 0.6493314676616916
 27%|██████████▊                             | 400/1473 [02:02<05:37,  3.17it/s]epoch 4 batch id 401 loss 0.9399889707565308 train acc 0.7191006857855362
 41%|████████████████▎                       | 600/1473 [03:02<04:22,  3.33it/s]epoch 4 batch id 601 loss 0.43351465463638306 train acc 0.7509879367720466
 54%|█████████████████████▋                  | 800/1473 [04:02<03:28,  3.22it/s]epoch 4 batch id 801 loss 0.8205083012580872 train acc 0.7753784332084894
 68%|██████████████████████████▍            | 1000/1473 [05:02<02:22,  3.32it/s]epoch 4 batch id 1001 loss 0.05804213881492615 train acc 0.7486107642357642
 81%|███████████████████████████████▊       | 1200/1473 [06:02<01:22,  3.31it/s]epoch 4 batch id 1201 loss 0.09878826886415482 train acc 0.7479704412989175
 95%|█████████████████████████████████████  | 1400/1473 [07:01<00:21,  3.37it/s]epoch 4 batch id 1401 loss 0.11601975560188293 train acc 0.7506245538900785
100%|███████████████████████████████████████| 1473/1473 [07:23<00:00,  3.32it/s]
epoch 4 train acc 0.7628139850644943
100%|███████████████████████████████████████| 1473/1473 [02:11<00:00, 11.20it/s]
epoch 4 test acc 0.06048455532926001

**Comment by Jeiyoon** [comment]
해결했습니다! loss spikes 문제였네요.

이 이슈는 닫겠습니다

다시한번 좋은 소스 오픈에 대해 감사말씀 드립니다.
